spec/lib/mvi/responses/find_candidate_spec.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+require 'rails_helper'
+require 'mvi/responses/find_candidate'
+require "#{Rails.root}/spec/support/mvi/mvi_response"
+
+describe MVI::Responses::FindCandidate do
+  context 'given a valid savon response' do
+    let(:valid_response) { MVI::Responses::FindCandidate.new(mvi_valid_response) }
+
+    describe '#invalid?' do
+      it 'should return false' do
+        expect(valid_response.invalid?).to be_falsey
+      end
+    end
+
+    describe '#failure?' do
+      it 'should return false' do
+        expect(valid_response.failure?).to be_falsey
+      end
+    end
+
+    describe '.body' do
+      context 'with middle name and icn, mhv correlation ids' do
+        it 'should filter the patient attributes the system is interested in' do
+          expect(valid_response.body).to eq(
+            birth_date: '19800101',
+            edipi: '1234^NI^200DOD^USDOD^A',
+            vba_corp_id: '12345678^PI^200CORP^USVBA^A',
+            family_name: 'Smith',
+            gender: 'M',
+            given_names: %w(John William),
+            icn: '1000123456V123456^NI^200M^USVHA^P',
+            mhv_id: '123456^PI^200MHV^USVHA^A',
+            ssn: '555443333',
+            status: 'active'
+          )
+        end
+      end
+
+      context 'when name parsing fails' do
+        it 'should set the names to false' do
+          allow_any_instance_of(String).to receive(:capitalize).and_raise(StandardError)
```

Comment:
This test is a bit confusing, if a response causes name parsing to fail, is it still valid? Stubbing String to raise an error on capitalize does not mimic a real scenario, can we use data that would cause the name parsing to fail (something that would call capitalize on nil or similar)?